charts/datadog/templates/agent-priorityclass.yaml

```diff
@@ -0,0 +1,9 @@
+{{- if and (or .Values.providers.gke.autopilot .Values.agents.priorityClassCreate) (not .Values.agents.priorityClassName)}}
+apiVersion: scheduling.k8s.io/v1
+description: Used for Datadog Agent Components to be scheduled with higher priority.
+kind: PriorityClass
+metadata:
+  name: {{ template "datadog.fullname" . }}
```

Comment:
What if the user want to set a specific name for the pc?
I think you can reuse the logic used in the pod template